Update plugin.py

artodrees opened the pull request:
Adds indirect parameterization support and fixture teardown grouping

artodrees committed:
Update plugin.py

Adds indirect parameterization support and fixture teardown grouping

artodrees committed:
Update plugin.py

artodrees committed:
Update test_e2e.py

artodrees committed:
Update plugin.py

artodrees committed:
Update test_e2e.py

artodrees committed:
Update plugin.py

artodrees committed:
store active group name and print it in endgroup command for debug

artodrees committed:
fix last commit

artodrees committed:
Update .gitignore

artodrees committed:
lint fixes

artodrees committed:
remove printing previously active group

jupe reviewed (commented):
not yet working properly: 

![A6354C1B-227A-4744-9B0C-0DE41A3FD35D](https://user-images.githubusercontent.com/3822425/147125075-a77c546d-315a-482b-89cd-40473125c153.jpeg)

jupe commented on source/plugin.py:
```diff
@@ -109,4 +112,11 @@ def pytest_fixture_setup(request, fixturedef) -> None:
+        """
+        Start group "FIXTURE FixtureName TEARDOWN
+        """
+        pytest.grouping_session.start_github_group(prefix=fixture_type, name=fixture_name,
```
this probably leads to invalid `::endgroup::` 🤔

artodrees committed:
fix parameter detection

artodrees committed:
parameter tests

artodrees committed:
ensure command is on new line

artodrees committed:
fix gh_log_group option

artodrees committed:
ensure newlines before and after commands

artodrees committed:
add "end_group" finalizer for fixtures

artodrees committed:
don't add newlines after commands

hnrhonkala88 committed:
Merge branch 'main' into patch-2

hnrhonkala88 committed:
Update plugin.py

parameters in wrong order..

hnrhonkala88 commented:
fixed merge conflict when merged master to this PR and fixed bug that came up during the merge conflict fix.

jupe commented:
still not handle lines properly:
<img width="371" alt="image" src="https://user-images.githubusercontent.com/3822425/148893731-99d01417-5597-41e4-929b-3a155a7a788d.png">

artodrees commented:
> still not handle lines properly: 
`<img>`

Yes, it's a pain with terminalreporter... 
i'm working on this atm.

artodrees committed:
try to force own hooks to be run first

Make most hook implementaions hookwrappers so they run as soon (or as late) as possible

artodrees committed:
trying to fix newlines

artodrees committed:
use f-string for github group name

artodrees committed:
pytest_runtest_logreport group closing after yield

artodrees committed:
fix ending groups

artodrees committed:
see how printing reports out from groups looks like

artodrees committed:
add extra checks for verbose, quiet, and emoji reporting

artodrees committed:
fix indentation

artodrees committed:
fix workflow

artodrees committed:
workflow tweaks

artodrees committed:
Revert "see how printing reports out from groups looks like"

This reverts commit 620d5324595945cb1ec4aa21f942820f8f46d4f5.

artodrees commented:
finally starting to look correct:
![image](https://user-images.githubusercontent.com/88309388/148908159-66340149-d73d-41d2-8208-80f6a72552ad.png)
